### PR TITLE
Added Username and Password for MQTT

### PR DESCRIPTION
**Pull request by nmaas87**
I added Username and Password for basic MQTT authentication for security reasons (so that we can at least use an basic protection on our server and the acl features of MQTT).

### CODE REVIEW

**Comment by stintel** [comment]
@nmaas87 hey, I see you added the username and password to the README.md, but in fact that's unrelated to the missing pkg-config dependency. This should be part of the other commit where you actually add these options.

Thanks for resubmitting btw. I'm exploring some options to avoid the previous issues in the future.

**Comment by nmaas87** [comment]
OK, I just repushed with the Commands in right order (grouped Username and Password) and extracted the other README.md change. I am going to make an PR as soon as you accepted this one, because if I try to do it now, it will inject it into this one.

**Comment by stintel** [comment]
Morning @nmaas87, looks like the author still doesn't match the signed-off-by line.
Author: Nico Maas <xenomorph150@gmail.com>
Signed-off-by: Nico Maas <mail@nico-maas.de>

Can you fix that? Tip: git config user.email mail@nico-maas.de, then git commit -s --amend --reset-author

Also not sure what you mean. I guess you should have a look at git format-patch and git am. I'm using that very often when improving PRs with >1 commit.

**Comment by nmaas87** [comment]
Morning @stintel, I just did it, should now match.

File: src/main.c [hunk 1]
```diff
@@ -28,21 +28,25 @@
 
 static void print_usage()
 {
-    printf("Usage: -h host [-p port] -t topic1 [-t topicN]\n");
+    printf("Usage: -h host [-p port] [-u username] [-P password] -t topic1 [-t topicN]\n");
     printf(" -h: MQTT host to connect to\n");
     printf(" -p: MQTT port to connect to (1883)\n");
+    printf(" -u: MQTT username\n");
+    printf(" -P: MQTT password\n");
```
Comment (stintel) [inline src/main.c]: You have trailing spaces here. Should be easy to spot when you run git diff before or git show after commit.

**Comment by nmaas87** [comment]
Fixed

**Comment by stintel** [comment]
Merge, thanks for your patience! :-)

**Pull request merged**